client/components/point-estimate/estimate-select.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useMutation, useQuery } from 'urql';
+
+import { Story } from '~components/projects/types';
+import { Estimate } from '~components/projects/types';
+import { ReduxState } from '~redux/reducers';
+
+import { setStory } from '~redux/actions/story';
+
+const getValidEstimates = `query GetValidEstimates($projectId: String!) {
+  pointingSchema(projectId: $projectId) {
+    ... on EstimateSchema {
+      all
+    }
+    ... on DbError {
+      message
+    }
+    ... on PtError {
+      code
+      error
+      kind
+      possibleFix
+    }
+  }
+}`;
+
+const addEstimate = `mutation AddStoryEstimate($storyId: String!, $pointValue: Int!, $userEmail: String!) {
+  createEstimate(storyId: $storyId, pointValue: $pointValue, userEmail: $userEmail) {
+    id
+    pointValue
+    userId
+  }
+}`;
+
+interface Props {
+  story: Story;
+}
+
+const EstimateSelect = ({ story }: Props) => {
+  const currentProject = useSelector((state: ReduxState) => state.project);
+  const currentUser = useSelector((state: ReduxState) => state.user);
+  const storyPosition = useSelector((state: ReduxState) => state.story.storyPosition);
+  const dispatch = useDispatch();
+
+  const [, executeMutation] = useMutation(addEstimate);
+
+  const onChange = e => {
+    const storyId = story.id;
+    executeMutation({
+      storyId,
+      pointValue: parseInt(e.target.value, 10),
+      userEmail: currentUser.email,
+    });
+    if (story.userEstimates.length) {
+      story.userEstimates[0].pointValue = e.target.value;
```

Comment:
Why do we mutates props? Can we assume this is always changing the first `userEstimates` element?